rero_ils/modules/acq_orders/serializers/csv.py

```diff
@@ -0,0 +1,283 @@
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Acquisition order serialization."""
+
+import csv
+
+from flask import current_app, request, stream_with_context
+from invenio_i18n.ext import current_i18n
+from invenio_records_rest.serializers.csv import CSVSerializer, Line
+
+from rero_ils.modules.acq_accounts.api import AcqAccountsSearch
+from rero_ils.modules.acq_order_lines.api import AcqOrderLinesSearch
+from rero_ils.modules.acq_receipt_lines.api import AcqReceiptLinesSearch
+from rero_ils.modules.acq_receipts.api import AcqReceiptsSearch
+from rero_ils.modules.documents.api import DocumentsSearch
+from rero_ils.modules.documents.serializers.base import DocumentFormatter
+from rero_ils.modules.vendors.api import VendorsSearch
+from rero_ils.utils import get_i18n_supported_languages
+
+creator_role_filter = [
+    'rsp', 'cre', 'enj', 'dgs', 'prg', 'dsr', 'ctg', 'cmp', 'inv', 'com',
+    'pht', 'ivr', 'art', 'ive', 'chr', 'aut', 'arc', 'fmk', 'pra', 'csl'
+]
+
+
+class AcqOrderCSVSerializer(CSVSerializer):
+    """Mixin serializing records as CSV."""
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        :param item_links_factory: Factory function for record links.
+        """
+        # define chunk size
+        chunk_size = 100
+        vendors = {}
+
+        # language
+        language = request.args.get("lang", current_i18n.language)
+        if not language or language not in get_i18n_supported_languages():
+            language = current_app.config.get('BABEL_DEFAULT_LANGUAGE', 'en')
+
+        order_fields = {
+            'order_pid': 'pid',
+            'order_reference': 'reference',
+            'order_date': 'order_date',
+            'order_type': 'type',
+            'order_status': 'status'
+        }
+
+        def generate_csv():
+            def batch(results):
+                """Chunk search results.
+
+                :param results: search results.
+                :return list of chunked item pids and search records
+                """
+                records = {}
+                pids = []
```

Comment:
VERY COSMETIC : you could use single line statement : `records, pids = {}, []`

Reply:
ok